app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt

```diff
@@ -893,6 +895,39 @@ class HomeActivityTest {
     }
   }
 
+  @Test
+  fun testHomeActivity_promotedStoryHasScalableWidth() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_UPTIME_MILLIS)
+    storyProgressTestHelper.markInProgressSavedFractionsStory0Exp0(
+      profileId = profileId1,
+      timestampOlderThanOneWeek = false
+    )
+    logIntoUserTwice()
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId1)).use {
+      it.onActivity { activity ->
+        testCoroutineDispatchers.runCurrent()
+        scrollToPosition(position = 1)
+        onView(
+          atPositionOnView(
+            recyclerViewId = R.id.home_recycler_view,
+            position = 0,
+            targetViewId = R.id.promoted_story_list_recycler_view
+          )
+        ).check { _, _ ->
+          val promotedStoryCard =
+            activity.findViewById<LessonThumbnailImageView>(R.id.lesson_thumbnail)
+          val pixels = promotedStoryCard?.width?.toFloat()
+          val expectedSize = TypedValue.applyDimension(
+            TypedValue.COMPLEX_UNIT_SP,
+            280F,
+            context.resources.displayMetrics
+          )
+          assertThat(pixels).isEqualTo(expectedSize)
```

Comment:
Float comparisons need to use a threshold to ensure that rounding errors don't cause inconsistent results. You can use a mid-sized epsilon for most cases, e.g.:

```kotlin
assertThat(pixels).isWithin(1e-5f).of(expectedSize)
```

Reply:
Updated